File: app/src/main/java/xyz/klinker/messenger/fragment/settings/MaterialPreferenceFragment.java
package xyz.klinker.messenger.fragment.settings;

import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.preference.PreferenceFragment;
import android.support.annotation.Nullable;
import android.view.View;
import android.widget.ListView;

import xyz.klinker.messenger.R;
import xyz.klinker.messenger.shared.util.DensityUtil;

/**
 * To be used with the MaterialPreferenceCategory
 */
public abstract class MaterialPreferenceFragment extends PreferenceFragment {

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        ListView list = (ListView) view.findViewById(android.R.id.list);
        list.setBackgroundColor(getResources().getColor(R.color.drawerBackground));
        list.setDivider(new ColorDrawable(getResources().getColor(R.color.background)));
        list.setDividerHeight(DensityUtil.INSTANCE.toDp(getActivity(), 1));
    }
}

File: app/src/main/java/xyz/klinker/messenger/fragment/settings/MaterialPreferenceFragment.kt
package xyz.klinker.messenger.fragment.settings

import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.preference.PreferenceFragment
import android.view.View
import android.widget.ListView

import xyz.klinker.messenger.R
import xyz.klinker.messenger.shared.util.DensityUtil

/**
 * To be used with the MaterialPreferenceCategory
 */
@Suppress("DEPRECATION")
abstract class MaterialPreferenceFragment : PreferenceFragment() {

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        val list = view.findViewById<View>(android.R.id.list) as ListView
        list.setBackgroundColor(resources.getColor(R.color.drawerBackground))
        list.divider = ColorDrawable(resources.getColor(R.color.background))
        list.dividerHeight = DensityUtil.toDp(activity, 1)
    }
}
